### PR TITLE
Detector FDD

### PR DESCRIPTION
**Pull request by mbroz84**
FDD is the AD upgrade. This commit consist of the basic detector geometry (only scintillator modules and wave length shifters) and simulation machinery up to hits. Code was tested via o2sim_serial using box generator in eta acceptance of the FDD (part of the commit), hits are created and looks healthy.

### CODE REVIEW

**Comment by sawenzel** [comment]
Thanks. Waiting for the codechecker report.

**Comment by sawenzel** [comment]
There are few codechecker errors:
```
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/build_o2checkcode_o2/0/Detectors/FIT/FDD/base/src/Geometry.cxx:116:31: error: use nullptr [modernize-use-nullptr]
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/build_o2checkcode_o2/0/Detectors/FIT/FDD/base/src/Geometry.cxx:145:30: error: use nullptr [modernize-use-nullptr]
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/build_o2checkcode_o2/0/Detectors/FIT/FDD/simulation/src/Detector.cxx:339:87: error: use nullptr [modernize-use-nullptr]
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/build_o2checkcode_o2/0/Detectors/FIT/FDD/simulation/src/Detector.cxx:349:88: error: use nullptr [modernize-use-nullptr]
```

File: Detectors/FIT/FDD/base/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+set(MODULE_NAME "FDDBase")
+
```
Comment (sawenzel) [inline Detectors/FIT/FDD/base/CMakeLists.txt]: Why is FDD part of the FIT directory?

Comment (mbroz84) [reply Detectors/FIT/FDD/base/CMakeLists.txt]: From the management point of view it will be part of FIT in the same way as T0/V0.

Comment (sawenzel) [reply Detectors/FIT/FDD/base/CMakeLists.txt]: Thanks for the info.

**Comment by mbroz84** [comment]
Is it possible to run the codechecker somehow locally before commit?

**Comment by sawenzel** [comment]
Yes. This is explained here: https://github.com/AliceO2Group/AliceO2/tree/dev/Utilities/Tools

**Pull request merged**